docs/DeveloperGuide.adoc

```diff
@@ -295,25 +311,90 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+
+|`* *` |busy person |enter the time as yesterday, one hour ago, etc |save effort by using convenient and flexible time formats
+
+|`* *` |non-money minded person |receive warnings when I’m close to or exceeding my budget |adjust my expenses to prevent overspending
+
+|`* *` |careless person who is prone to typos |undo my previous command |I do not enter the wrong amount for my expenses
+
+|`* *` |careless person who is prone to typos |redo my previous command |I do not delete expenses by accident
+
+|`* *` |non-money minded person |set periodic budgets |discipline my spending
+
+|`* *` |non-money minded person |set categorical budgets |exceptional expenses (e.g. healthcare or sponsored expenses) are not tracked under default budget to prevent confusion
+
+|`* *` |non-money minded person |let the App suggest some budgeting tips for me |be better at tracking my expenses
+
+|`*` |forgetful person |receive email reminders if I haven’t input my expenses for a few days |do not forget to track my expense
+
+|`*` |someone who spends a lot of time in public spaces in school |easily hide my budgeting information on the application quickly |others could not see the details on my screen
+
+|`*` |busy person |input my expenses by scanning a receipt |can save more time than manually typing in the expenses
 
-|`* * *` |user |add a new person |
+|`*` |NUS student, and lazy person |add expense from school canteens by saying e.g. “deck steamed chicken rice” |don't need to enter the details and prices
 
-|`* * *` |user |delete a person |remove entries that I no longer need
+|`*` |non-financially independent student |hide some details when exporting expenses |do not share exact details with my parents but instead include them as an anonymous expense
 
-|`* * *` |user |find a person by name |locate details of persons without having to go through the entire list
+|`*` |non-money minded person |let the App help me make saving plans for future expenses |start saving early
 
-|`* *` |user |hide <<private-contact-detail,private contact details>> by default |minimize chance of someone else seeing them by accident
+|`*` |user |group certain expenses, e.g. group chicken rice and milo as lunch, but both remain separate entries as well |manage related expenses together
+
+|`*` |advanced CLI user |add aliases to my common inputs |avoid typing the similar things every time
+
+|`*` |user |let the App predict frequent similar expenses and prompt me to enter them |do not miss expenses while I'm not using the App
+
+|`*` |busy student who may need to close the App to settle some other matters |let the App remember what I had typed into the command line last time I closed the App |resume when I reopen the app
 
-|`*` |user with many persons in the address book |sort persons by name |locate a person easily
 |=======================================================================
 
 _{More to be added}_
 
 [appendix]
 == Use Cases
```

Comment:
*Use Cases Review*

1) Step numbering is accurate. Well Done!
2) Some possible extensions mentioned. Well Done!
3) No unnecessary UI details mentioned. Well Done!
4) Formatting and notations are accurate. Well Done!